xla/service/gpu/runtime/copy_thunk.h

```diff
@@ -55,22 +59,70 @@ class DeviceToDeviceCopyThunk : public Thunk {
   const uint64_t mem_size_;
 };
 
-class DeviceToHostCopyThunk : public DeviceToDeviceCopyThunk {
- public:
-  DeviceToHostCopyThunk(ThunkInfo thunk_info,
-                        const BufferAllocation::Slice& source_buffer,
-                        const BufferAllocation::Slice& destination_buffer,
-                        uint64_t mem_size);
-  absl::Status ExecuteOnStream(const ExecuteParams& params) override;
+//===----------------------------------------------------------------------===//
+// CopyAsyncEvents
+//===----------------------------------------------------------------------===//
+class CopyAsyncEvents {
```

Comment:
When you'll have a `CopyThunk` (see another comment) you an make this a member class and name it simply `AsyncEvents` (`CopyThunk::AsyncEvents`)

Reply:
Yes, made it CopyThunk::AsyncEvents.